backend/dal/Helpers/Extensions/QueryableExtensions.cs

```diff
@@ -0,0 +1,161 @@
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace Pims.Dal.Helpers.Extensions
+{
+    /// <summary>
+    /// QueryableExtensions static class, provides extension methods for IQueryable.
+    /// </summary>
+    public static class QueryableExtensions
+    {
+        #region Variables
+        private static readonly string[] _orderByDescending = new[] { "desc", "descending" };
+        private static readonly MethodInfo OrderByMethod = typeof(Queryable).GetMethods().Single(method => method.Name == "OrderBy" && method.GetParameters().Length == 2);
+        private static readonly MethodInfo OrderByDescendingMethod = typeof(Queryable).GetMethods().Single(method => method.Name == "OrderByDescending" && method.GetParameters().Length == 2);
+        private static readonly MethodInfo GeneratePropertyPathLambdaMethod = typeof(QueryableExtensions).GetMethod(nameof(GeneratePropertyPathLambda), BindingFlags.NonPublic | BindingFlags.Static);
+        #endregion
+
+        /// <summary>
+        /// Check if the specified property exists in the specified type.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="propertyName"></param>
+        /// <returns></returns>
+        private static bool PropertyExists<T>(string propertyName)
+        {
+            return typeof(T).GetProperty(propertyName, BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.Instance) != null;
+        }
+
+        /// <summary>
+        /// Order the query results by the specified property names.
+        /// Each property can also specify the direction of the sort (i.e. "Name asc", "Name ascending", "Name desc", "Name descending").
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="source"></param>
+        /// <param name="propertyName"></param>
+        /// <returns></returns>
+        public static IQueryable<T> OrderByProperty<T>(this IQueryable<T> source, params string[] propertyName)
```

Comment:
Provides a way to dynamically handle sorting by an array of property path values.